app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts

```diff
@@ -0,0 +1,215 @@
+        return;
+      }
+      if (code === 91 || code === 92 || code === 93) { // left + right command/window, select
+        return;
+      }
+      if (code > 111 && code < 186) { // f keys, misc
+        return;
+      }
+    }
+    this.executeQuery();
+
+  }
+
+  public focusFirstSearchResult(event: any) {
+    try {
+      event.preventDefault();
+      this.$element.find('ul.dropdown-menu').find('a').first().focus();
+    } catch (e) {
+      this.$log.debug(e);
+    }
+  }
+
+  public searchFieldBlurred(blurEvent: any) {
+    // if the target is outside the global search (e.g. shift+tab), hide the results
+    if (!$.contains(this.$element.get(0), blurEvent.relatedTarget)) {
+      this.hideResults();
+    }
+  }
+
+  private focusLastSearchResult(event: any) {
+    try {
+      event.preventDefault();
+      this.$element.find('ul.dropdown-menu').find('a').last().focus();
+    } catch (e) {
+      this.$log.debug(e);
+    }
+  }
+
+  private executeQuery() {
+    if (this.query) {
+      this.querying = true;
+      this.filteredCategories = _.compact(this.categories.map((category: any) => {
+        let results = category.results.filter((result: string) => result.toLowerCase().includes(this.query.toLowerCase()));
+        if (results.length > 0) {
+          return {category: category.category, results: results};
+        }
+      }));
+    }
+    this.querying = false;
+    this.showSearchResults = !!this.query;
+  }
+
+
+  private createFilterCategories(properties: FastProperty[]) {
+    this.categories = properties.reduce((acc: any, property: FastProperty) => {
+      let scope: FastPropertyScope = property.scope;
+      acc = this.addToList(acc, 'app', scope.app );
+      acc = this.addToList(acc, 'env', scope.env );
+      acc = this.addToList(acc, 'region', scope.region );
+      acc = this.addToList(acc, 'stack', scope.stack);
+      acc = this.addToList(acc, 'cluster', scope.cluster);
+      return acc;
+    }, []);
+
+  }
+
+  private addToList(acc: any[], scopeKey: string, scopeValue: string): any {
+    if (scopeValue) {
+      let categoryIndex = _.findIndex(acc, ['category', scopeKey]);
+      if (categoryIndex > -1) {
+        acc[categoryIndex].results = Array.from(new Set(acc[categoryIndex].results).add(scopeValue));
+      } else {
+        acc.push({category: scopeKey, results: ['none', scopeValue]});
+      }
+    }
+    return acc;
+  }
+}
+
+
+
+class FastPropertyFilterSearchDirective implements ng.IComponentOptions {
+
+  public bindings: any = {
+    'properties': '=',
+    'filters': '=',
+    'createFilterTag': '='
+  };
+  public controller: ng.IComponentController = FastPropertyFilterSearchController;
+  public controllerAs: string = 'fpFilter';
+  public templateUrl: string = require('./fastPropertyFilterSearch.component.html');
+
+}
+
+
+const moduleName = 'spinnaker.netflix.fastPropertyFilterSearch.directive';
```

Comment:
prefer exporting this, e.g.
```javascript
export const FAST_PROPERTY_SEARCH_COMPONENT = 'spinnaker.netflix.fastPropertyFilterSearch.component';
```
then doing an explicit import on the field where required